rasa/core/policies/intent_ted_policy.py

```diff
@@ -0,0 +1,691 @@
+        """
+        label_data = RasaModelData()
+        label_data.add_data(attribute_data, key_prefix=f"{LABEL_KEY}_")
+        label_data.add_lengths(
+            f"{LABEL}_{INTENT}", SEQUENCE_LENGTH, f"{LABEL}_{INTENT}", SEQUENCE,
+        )
+        label_ids = np.arange(len(domain.intents))
+        label_data.add_features(
+            LABEL_KEY,
+            LABEL_SUB_KEY,
+            [FeatureArray(np.expand_dims(label_ids, -1), number_of_dimensions=2)],
+        )
+        return label_data
+
+    @staticmethod
+    def _prepare_data_for_prediction(model_data: RasaModelData) -> RasaModelData:
+        """Transform training model data to data usable for making model predictions.
+
+        Transformation involves filtering out all features which
+        are not useful at prediction time. This is important
+        because the prediction signature will not contain these
+        attributes and hence prediction will break.
+
+        Args:
+            model_data: Data used during model training.
+
+        Returns:
+            Transformed data usable for making predictions.
+        """
+        filtered_data: Dict[Text, Dict[Text, Any]] = {
+            key: features
+            for key, features in model_data.data.items()
+            if key in PREDICTION_FEATURES
+        }
+        return RasaModelData(data=filtered_data)
+
+    def run_post_training_procedures(
+        self, model_data: RasaModelData, label_ids: np.ndarray
+    ) -> None:
+        """Run any post training tasks.
+
+        Args:
+            model_data: Data used for training the model.
+            label_ids: Numerical IDs of labels for each data point used during training.
+        """
+        # `model_data` contains data attributes like `label` which were
+        # used during training. These attributes are not present in
+        # the `predict_data_signature`. Prediction through the model
+        # will break if `model_data` is passed as it is through the model.
+        # Hence, we first filter out the attributes inside `model_data`
+        # to keep only those which should be present during prediction.
+        model_prediction_data = self._prepare_data_for_prediction(model_data)
+        self.label_thresholds = self.model.compute_thresholds(
+            model_prediction_data, label_ids
+        )
+
+    def _collect_action_metadata(
+        self, domain: Domain, similarities: np.array
+    ) -> Dict[Text, Dict[Text, float]]:
+        """Add any metadata to be attached to the predicted action.
+
+        Similarities for all intents and their thresholds are attached as metadata.
+
+        Args:
+            domain: Domain of the assistant.
+            similarities: Predicted similarities for each intent.
+
+        Returns:
+            Metadata to be attached.
+        """
+        metadata = {}
+        for intent in domain.intents:
+            intent_index = domain.intents.index(intent)
```

Comment:
```suggestion
        for intent_index, intent in enumerate(domain.intents):
```
Unless we have some reason to believe the order might change?